qiskit/primitives/backend_sampler.py

```diff
@@ -151,15 +151,29 @@ def _call(
         bound_circuits = self._bound_pass_manager_run(bound_circuits)
 
         # Run
-        result = self._backend.run(bound_circuits, **run_options).result()
+        if isinstance(self._backend, BackendV1):
+            max_circuits = self._backend.configuration().max_experiments
+        elif isinstance(self._backend, BackendV2):
+            max_circuits = self._backend.max_circuits
+        if max_circuits:
+            jobs = [
+                self._backend.run(bound_circuits[pos : pos + max_circuits], **run_options)
+                for pos in range(0, len(bound_circuits), max_circuits)
+            ]
+            result = [x.result() for x in jobs]
+        else:
+            result = [self._backend.run(bound_circuits, **run_options).result()]
 
         return self._postprocessing(result, bound_circuits)
 
     def _postprocessing(self, result: Result, circuits: list[QuantumCircuit]) -> SamplerResult:
 
-        counts = result.get_counts()
-        if not isinstance(counts, list):
-            counts = [counts]
+        counts = []
```

Comment:
This part is a counterpart of job splitting. if you make a utility function, you can use it for both estimator and sampler. What do you think?